src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java

```diff
@@ -0,0 +1,142 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+
+
+public class Post extends BasePageObject {
+
+  @FindBy(css = ".post-detail")
+  private List<WebElement> postList;
+
+  @FindBy(css = ".user-avatar")
+  private WebElement avatarImage;
+
+  @FindBy(css = ".avatar-username")
+  private WebElement avatarUsername;
+
+  @FindBy(css = ".icon.upvote")
+  private List<WebElement> replyUpvoteButton;
+
+  @FindBy(css = ".upvote-area")
+  private List<WebElement> replyVoteCount;
+
+  @FindBy(css = ".toggle-share")
+  private List<WebElement> toggleShare;
+
+  @FindBy(css = ".share-feature")
+  private List<WebElement> shareFeature;
+
+  @FindBy(css = "a.upvote-area")
+  private WebElement upvoteArea;
+
+  @FindBy(css = "svg.upvote")
+  private WebElement upvoteButton;
+
+  public boolean isPostListEmpty() {
+    return postList.isEmpty();
+  }
+
+  public Post clickUserAvatar() {
+    avatarImage.click();
+    return this;
+  }
+
+  public Post clickUsernameLink() {
+    avatarUsername.click();
+    return this;
+  }
+
+  public Post scrollToBottom(WebDriver driver) {
+    ((JavascriptExecutor) driver)
+        .executeScript("window.scrollTo(0, document.body.scrollHeight)");
+    return this;
+  }
+
+  public int getPostsListLength() {
+    return postList.size();
+  }
+
+  public boolean isUpvoteButtonVisible(int index) {
+    WebElement button = replyUpvoteButton.get(index);
+    wait.forElementVisible(button);
+    return button.isDisplayed();
+  }
+
+  public String getVoteCount(int index) {
+    WebElement voteCountArea = replyVoteCount.get(index);
+    wait.forElementVisible(voteCountArea);
+    return voteCountArea.getText();
+  }
+
+  public void clickUpvoteButton(int postIndex) {
+    WebElement button = replyUpvoteButton.get(postIndex);
+    wait.forElementClickable(button);
+    button.click();
+  }
+
+  public void waitForVoteCountToChange(int postIndex, String voteCount) {
+    WebElement voteArea = replyVoteCount.get(postIndex);
+    wait.forTextNotInElement(voteArea, voteCount);
+  }
+
+  /**
+   * Wait for the noticeable time lag between vote and vote value change to pass
+   */
+  public void waitForVoteCountChangeTimeLagToPass() {
+    try {
+      //This sleep was introduced because of noticeable lag between vote and vote value change
+      Thread.sleep(2000);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1148)